resources/common.rb

```diff
@@ -0,0 +1,34 @@
+actions :install
+default_action :install
+
+# Name the resource
+attribute :name, :name_attribute => true, 'kind_of' => String,
+                 :required => true
+
+# Creates the user if they do not already exist
+attribute :create_user, 'kind_of' => [TrueClass, FalseClass], :default => false
+
+# Create the owner, path, or group if they do not exist
+# If path is nil it will default to '/opt/<name_attribute>'
+attribute :path, 'kind_of' => [String, nil], :default => nil
+attribute :owner, 'kind_of' => String, :default => 'chef'
+attribute :group, 'kind_of' => String, :default => 'chef'
+
+# The information necessary to check out the code
+attribute :repository, 'kind_of' => String
+attribute :revision, 'kind_of' => String, :default => 'master'
+
+# Location of the virtualenv
+attribute :virtualenv_path, 'kind_of' => String, :default => '/opt/venv'
```

Comment:
How about `/opt/venv_$PROJECT_NAME` and have this nil or String nonsense further infect the project?